src/v/cluster/controller_backend.cc

```diff
@@ -657,19 +657,28 @@ controller_backend::calculate_learner_initial_offset(
       std::min(max_removable_local_log_offset, *retention_offset));
 }
 
-void controller_backend::process_delta(const topic_table::ntp_delta& d) {
+void controller_backend::process_deltas(
+  chunked_vector<topic_table::ntp_delta> deltas) {
+    ssx::spawn_with_gate(_gate, [this, deltas = std::move(deltas)] mutable {
+        return ss::do_with(
+          std::move(deltas),
+          [this](chunked_vector<topic_table::ntp_delta>& deltas) {
+              return ss::max_concurrent_for_each(
+                deltas, 1024, [this](topic_table::ntp_delta d) {
```

Comment:
The 1024 are way too large here I think. You get the task queue warnings already at 1000 elements so you'd instead probably rather want something like 10 or 100.